recipe/meta.yaml

```diff
@@ -55,23 +55,25 @@ requirements:
         - geos
         - proj.4
         - xerces-c
-        - libnetcdf
+        - libnetcdf 4.3*
         - krb5
-        - libtiff
         - libpng
-        - openjpeg
+        - openjpeg  # [not py35]
+        - libtiff 4.0.6  # [not py35]
```

Comment:
> In what way? Is the pinned version wrong, or is the defaults package wrong? 

I am not sure if the default package is broken or if it is not playing nice with our packages. `libtiff` depends on `zlib`, `xz`, and `jpeg` (all are properly pinned now).  Even if it is pinning I don't get why conda is installing an older version of `libtiff` there.

Check this [log](https://travis-ci.org/conda-forge/gdal-feedstock/jobs/124339146) where you can see that the default channel `libtiff 4.0.2` is [picked](https://travis-ci.org/conda-forge/gdal-feedstock/jobs/124339146#L320) instead of `4.0.6` when we pin with `4*`.

And that results in:

```shell
===== testing package: gdal-1.11.4-np110py34_5 =====
import: 'osr'
Traceback (most recent call last):
  File "/Users/travis/miniconda3/conda-bld/test-tmp_dir/run_test.py", line 25, in <module>
    import osr
  File "/Users/travis/miniconda3/envs/_test/lib/python3.4/site-packages/osr.py", line 2, in <module>
    from osgeo.gdal import deprecation_warn
  File "/Users/travis/miniconda3/envs/_test/lib/python3.4/site-packages/osgeo/__init__.py", line 21, in <module>
    _gdal = swig_import_helper()
  File "/Users/travis/miniconda3/envs/_test/lib/python3.4/site-packages/osgeo/__init__.py", line 17, in swig_import_helper
    _mod = imp.load_module('_gdal', fp, pathname, description)
  File "/Users/travis/miniconda3/envs/_test/lib/python3.4/imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
ImportError: dlopen(/Users/travis/miniconda3/envs/_test/lib/python3.4/site-packages/osgeo/_gdal.so, 2): Library not loaded: /opt/anaconda1anaconda2anaconda3/lib/libtiff.5.dylib
  Referenced from: /Users/travis/miniconda3/envs/_test/lib/libgdal.1.dylib
  Reason: image not found
TESTS FAILED: gdal-1.11.4-np110py34_5
```